tls/src/lib.rs

```diff
@@ -139,9 +147,17 @@ impl<T> Stream<T> {
         &mut self.inner
     }
 
-    pub fn peer_platform_id(&self) -> &PlatformId {
+    pub fn peer_platform_id_unchecked(&self) -> &PlatformId {
         &self.platform_id
     }
+
+    pub fn peer_platform_id(&self) -> Option<&PlatformId> {
+        if self.corpus_appraisal_success {
+            Some(&self.platform_id)
+        } else {
+            None
+        }
+    }
```

Comment:
I will admit this is a little goofy but I wanted to make it difficult to use the unappraised version without being intentional.

Reply:
Got it. I had a different expectation for how we'd always enable appraisal, which I believe is what's causing some confusion here. I figured that once we enabled appraisal, we'd do so by changing the sprockets code to return an error rather than making the caller check via looking at the platform id. 

In essence I expected we'd back out this code and go back to returning an error, rather than changing the code at the call sites. Once we have some testing under our belt, the API should not allow a stream to be constructed with a failure ever again. This is how SPDM works, and how I always intended sprockets to work. It makes it foolproof to do the right thing, because you literally can't get a stream object if appraisal fails.

Reply:
Okay I see your point. My thought was we could have some intermediate APIs but you're right we eventually want the current behavior.